pkg/core/resources/model/resource.go

```diff
@@ -458,15 +458,42 @@ func resourceOrigin(labels map[string]string) (mesh_proto.ResourceOrigin, bool)
 	return "", false
 }
 
+// Namespace type allows to avoid carrying both 'namespace' and 'systemNamespace' around the code base
+// and depend on this type instead
+type Namespace struct {
+	value  string
+	system bool
+}
+
+func NewNamespace(value string, system bool) Namespace {
+	return Namespace{
+		value:  value,
+		system: system,
+	}
+}
+
+func UnknownNamespace() Namespace {
+	return Namespace{}
+}
```

Comment:
Should the UnknownNamespace just be a constant? Makes it either to just check `if namespace == UnknownNamespace`

Reply:
It's also probably Undefined not Unknown. To me `Unkown` could be something that exists but we don't know what it is. Whereas `Undefined` just doesn't exist.

Btw in any case this and the method after deserve a little comment

Reply:
`Unset` is another option that might be even better

Reply:
Right, it's better as a constant, also changed the name to `UnsetNamespace`